src/stronghold.rs

```diff
@@ -137,6 +138,44 @@ pub async fn set_password_clear_interval(interval: Duration) {
     *clear_interval = interval;
 }
 
+/// Snapshot status.
+#[derive(Debug, Serialize)]
+#[serde(tag = "status", content = "data")]
+pub enum SnapshotStatus {
+    /// Snapshot is locked. This means that the password must be set again.
+    Locked,
+    /// Snapshot is unlocked. The duration is the amount of time left before it locks again.
+    Unlocked(Duration),
+}
+
+#[derive(Debug, Serialize)]
+pub struct Status {
+    snapshot: SnapshotStatus,
+}
+
+/// Gets the stronghold status for the given snapshot.
+pub async fn get_status(snapshot_path: &PathBuf) -> Status {
+    let password_clear_interval = *PASSWORD_CLEAR_INTERVAL
+        .get_or_init(|| Arc::new(Mutex::new(DEFAULT_PASSWORD_CLEAR_INTERVAL)))
+        .lock()
+        .await;
+    let access_interval = STRONGHOLD_ACCESS_STORE
+        .get_or_init(Default::default)
+        .lock()
+        .await
+        .get(snapshot_path)
+        .cloned()
+        .unwrap_or_else(Instant::now);
```

Comment:
Return older value, should be locked in case there is a problem

Reply:
done :) if it's None it'll just return Locked